pkgs/development/compilers/open-watcom-bin/default.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenvNoCC, fetchurl, qemu, expect, writeScriptBin, ncurses }:
+
+let
+
+  # We execute the installer in qemu-user, because otherwise the
+  # installer fails to open itself due to a failed stat() call. This
+  # seems like an incompatibility of new Linux kernels to run this
+  # ancient binary.
+  installExpectScript = writeScriptBin "perform-ow-install" ''
+    #!${expect}/bin/expect -f
+
+    spawn env TERMINFO=${ncurses}/share/terminfo TERM=vt100 qemu-i386 [lindex $argv 0]
```

Comment:
As `expect` and `ncurses` are addressed via their absolute store paths, i would find it cleaner if the same thing happened with qemu, too, instead of assuming that it comes from the environment. 

If then the `installExpectScript` would be a `writeScript`, it could be directly called like `${installExpectScript} install-bin` and we would not need any `nativeBuildInputs` any longer. I would prefer that just like i prefer to limit the scope of variables in normal programming.